blaseball_mike/events.py

```diff
@@ -16,14 +18,29 @@ async def stream_events(url='https://api.blaseball.com/events/streamData', retry
     `retry_max` is the maximum time to delay if there's a connection error
     """
     retry_delay = retry_base
+    event_current = {}
+    delta_previous = None
     while True:
         try:
-            async with sse_client.EventSource(url, read_bufsize=2 ** 18) as src:
+            async with sse_client.EventSource(url, read_bufsize=2 ** 19) as src:
                 async for event in src:
                     retry_delay = retry_base  # reset backoff
                     if not event.data:
                         continue
-                    payload = ujson.loads(event.data)['value']
+                    raw_event = ujson.loads(event.data)
+                    if 'value' in raw_event.keys(): # New, full event
+                        delta_previous = None
+                        event_current = raw_event['value']
+                        payload = event_current
+                    elif 'delta' in raw_event.keys(): # Delta event
+                        if raw_event['delta'] == delta_previous:
+                            continue
+                        delta_previous = raw_event['delta']
+                        jsonpatch.apply_patch(event_current, raw_event['delta'], in_place=True)
+                        payload = event_current
+                    else:
+                        print("Unknown event type.")
```

Comment:
let's add a param to this generator that's like `on_parse_error` that can maybe take in `"SKIP"` (silently continue), `"LOG"` (what you're doing now, maybe have it also pprint `raw_event` for dev purposes. Default mode), and `"RAISE"` (raise some `Exception` and kill the generator) and check it in here instead of just printing